### PR TITLE
feat(examples): Add metering example

### PR DESCRIPTION
**Pull request by jubianchi**
# To DO

- [x] How to use metering in a module
- [x] How to meter a function call (see cost of gas before and after)
- [x] Demonstrate that if the gas is over the given limit, it should fail.

# Review

- [x] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

File: examples/metering.rs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    // Create a Store.
+    // Note that we don't need to specify the engine/compiler if we want to use
+    // the default provided by Wasmer.
+    // You can use `Store::default()` for that.
+    let metering = Metering::new(5, |operator| -> u64 {
+        match operator {
+            Operator::LocalGet { .. }
+            | Operator::I32Add { .. }=> 2,
+            _ => {
+                dbg!(operator);
```
Comment (Hywan) [inline examples/metering.rs:51]: Is it going to be removed?

Comment (jubianchi) [reply examples/metering.rs:51]: yes it will, this patch is basically a draft/wip just so I don't loose the code in my stash or somewhere ;)

**Review by syrusakbary — COMMENTED** [review]
The metering example should show:

1. How to use metering in a module (done)
2. How to meter a function call (see cost of gas before and after)
3. Demonstrate that if the gas is over the given limit, it should fail.

**Review by syrusakbary — APPROVED** [review]

File: examples/metering.rs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+    println!(
+        "Remaining points: {:?}",
+        metering.get_remaining_points(&instance)
+    );
+    Ok(())
+}
+
+#[test]
+fn test_exported_function() -> Result<(), Box<dyn std::error::Error>> {
```
Comment (jubianchi) [inline examples/metering.rs:133]: I'll change that

**Pull request merged**